README.md

```diff
@@ -80,129 +80,260 @@ Mac : `source ./bin/set-environment-variables.sh`
 Once the containers are running, CCD's frontend can be accessed at [http://localhost:3451](http://localhost:3451).
 
-However, 6 more steps are required to correctly configure SIDAM and CCD before it can be used:
+---
+**NOTE**
+
+The `idam-api` container can be slow to start - both the `definition-store-api` and `data-store-api` containers will
+try to connect to the `idam-api` container when they start.
+
+If `idam-api` is not up and running and accepting connections
+you may see errors in the `definition-store-api` and `data-store-api` containers, such as
+
+```bash
+Caused by: org.springframework.web.client.ResourceAccessException: 
+    I/O error on GET request for "http://idam-api:5000/o/.well-known/openid-configuration": Connection refused (Connection refused);
+        nested exception is java.net.ConnectException: Connection refused (Connection refused)
+```
+
+If you the containers fail to start with these error, ensure `idam-api` is running using
+
+ ```bash
+curl http://localhost:5000/health
+ ```
+
+ensuring the response is 
+
+```bash
+{"status":"UP"}
+```
+
+Then restart the `definition-store-api` & `data-store-api` containers
+
+```bash
+./ccd compose restart ccd-definition-store-api ccd-data-store-api
+```
+---
+
+However, some more steps are required to correctly configure SIDAM and CCD before it can be used:
+
+---
+**NOTE**
+
+All scripts require the following environment variables to be set
+
+```bash
+IDAM_ADMIN_USER
+IDAM_ADMIN_PASSWORD
+```
+
+with the corresponding values from the confluence page at https://tools.hmcts.net/confluence/x/eQP3P
+
+### CCD Quick Start
+
+At this point most users can run the following 3 scripts 
+
+```bash
+./bin/add-idam-clients.sh
+./bin/add-roles.sh
+./bin/add-users.sh
+```
+
+to get their IDAM environment ready and then move on to the [Ready for take-off](###Ready-for-take-off) section.
+
+A more in depth explanation of the scripts is detailed below 
 
 ### 1. Configure Oauth2 Client of CCD Gateway on SIDAM
 
 An oauth2 client should be configured for ccd-gateway application, on SIDAM Web Admin.
+
+A script is provided that sets up the CCD Gateway client.  Execute the following:
+
+```bash
+./bin/add-idam-clients.sh
+```
+
+You may verify the service has been added by logging in to the SIDAM Web Admin with the URL and 
+logic credentials here: 
+
+https://tools.hmcts.net/confluence/x/eQP3P
+
+Navigate to 
+
+`Home > Manage Services`
+
+Optionally - to add any further IDAM service clients you can update the 
+
+```bash
+./bin/add-idam-clients.sh
+```
+
+to add a new entry and re-run the script (any entries in this file that already exist are skipped)
+
+`${dir}/utils/idam-create-service.sh LABEL CLIENT_ID CLIENT_SECRET REDIRECT_URL SELF_REGISTRATION SCOPE`
+
+---
+**NOTE**
+
+* SELF_REGISTRATION - a boolean parameter, defaults to a value of "false" if omitted
+* SCOPE - a space delimited string parameter, defaults to a value of "openid profile authorities acr roles search-user" if omitted
```

Comment:
default should be "openid profile roles" (to inline with IDAM).

Reply:
https://github.com/hmcts/ccd-docker/pull/103/commits/2a11ab59294f83f9ce45b5e51cda3ea2bc309264